stdpopsim/slim_engine.py

```diff
@@ -1833,6 +1839,41 @@ def _get_msp_rate_map(breaks, is_this_dfe, rate):
                 )
             )
 
+        # If haploid, rebuild individual table so each sample is an individual
+        if contig.ploidy == 1:
+
+            def to_ragged(x, idx):
+                return [x[idx[i] : idx[i + 1]] for i in range(len(idx) - 1)]
```

Comment:
Hm, isn't this related to `tskit.unpack_strings( )` (see [here](https://tskit.dev/tskit/docs/stable/python-api.html#tskit.unpack_bytes))? I guess it's not the same thing, and you probably tried that.